var/spack/repos/builtin/packages/py-plotnine/package.py

```diff
@@ -10,20 +12,44 @@ class PyPlotnine(PythonPackage):
     based on ggplot2. The grammar allows users to compose plots by explicitly
     mapping data to the visual objects that make up the plot."""
 
-    pypi = "plotnine/plotnine-0.8.0.tar.gz"
-
-    version(
-        "0.8.0",
-        sha256="39de59edcc28106761b65238647d0b1f6212ea7f3a78f8be0b846616db969276",
-    )
-
-    depends_on('python@3.6:', type=('build', 'run'))
-    depends_on("py-descartes@1.1.0:", type=("build", "run"))
-    depends_on("py-matplotlib@3.1.1:", type=("build", "run"))
-    depends_on("py-mizani@0.7.3:", type=("build", "run"))
-    depends_on("py-numpy@1.19.0:", type=("build", "run"))
-    depends_on("py-pandas@1.1.0:", type=("build", "run"))
-    depends_on("py-patsy@0.5.1:", type=("build", "run"))
-    depends_on("py-scipy@1.5.0:", type=("build", "run"))
-    depends_on("py-setuptools", type="build")
-    depends_on("py-statsmodels@0.12.1:", type=("build", "run"))
+    homepage = "https://plotnine.readthedocs.io/en/stable"
+    pypi     = "plotnine/plotnine-0.8.0.tar.gz"
+
+    version('0.9.0', sha256='0e89a93015f3c71d6844ac7aa9fb0da09b908f5f7dfa7dd5d68a5ca32b2ebcea')
+    version('0.8.0', sha256='39de59edcc28106761b65238647d0b1f6212ea7f3a78f8be0b846616db969276')
+    version('0.7.1', sha256='02f2b0435dae2e917198c5367fd97b010445d64d9888c6b7e755d3cdfe7ad057')
+    version('0.7.0', sha256='8ee67cbf010ccea32670760e930b7b02177030a89ccdf85e35d156a96ce36cd3')
+    version('0.6.0', sha256='aae2c8164abb209ef4f28cab01132d23f6879fcf8d492657487359e1241459e5')
+
+    depends_on('python@3.8:',            type=('build', 'run'), when='@0.9.0:')
+    depends_on('python@3.6:',            type=('build', 'run'), when='@0.6.0:')
+
+    depends_on("py-setuptools@59:",      type=("build", "run"), when='@0.9.0:')
+    depends_on("py-setuptools",          type=("build", "run"), when='@0.6.0:')
```

Comment:
I don't think this is needed at run-time. You're also missing a build dep on `py-setuptools-scm@6.4:+toml`